runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillStateInternals.java

```diff
@@ -474,9 +474,12 @@ private WindmillValue(
 
     @Override
     public void clear() {
-      modified = true;
+      // if the value was already null (because it was already cleared) then there's no need to
+      // mark this as modified again.  This will save having to persist a clear that does nothing.
```

Comment:
Wouldn't two calls to `clear()` toggle `modified` from `true` to `false`?

I think you would need to keep track of whether the value state is already in a cleared state instead of using ` value == null` as a proxy.

This logic could really use unit tests to validate such simple scenarios.

Reply:
good catch, I didn't really need to include this in here anyways, I'll revert it.